test/test_nn.py

```diff
@@ -7718,14 +7811,16 @@ def get_a_test_layer(use_cuda, activation, batch_first=False):
         use_cuda = torch.cuda.is_available()
         device = torch.device("cuda" if use_cuda else "cpu")
 
-        for batch_first in (True, False):
+        def _test(batch_first, training):
             def perm_fn(x):
                 return x.transpose(1, 0) if batch_first else x
 
             encoder_layer = get_a_test_layer(use_cuda=use_cuda, activation=activation,
                                              batch_first=batch_first)
 
             model = nn.TransformerEncoder(encoder_layer, 1).to(device)
+            if not training:
+                model = model.eval()
```

Comment:
you can do `mode.train(trainigng)`.
Also no need to get the result, this is inplace.